### PR TITLE
Add ruff format to CI

### PR DESCRIPTION
**Pull request by ultrabear**
Modified based off the pylint default template, uses py3.11 since that is what we work with

### CODE REVIEW

**Comment by ultrabear** [comment]
Awesome pip caching is a hack

**Comment by ultrabear** [comment]
Hmm, I wonder if putting poetry in requirements.txt could work

**Comment by ultrabear** [comment]
Looks like its marginally faster, but not even by much

File: .github/workflows/format.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: Ruff Format
+
+on: [push]
```
Comment (proegssilb) [inline .github/workflows/format.yml:3]: Does this also handle PRs?

File: requirements.txt
Comment (proegssilb) [inline requirements.txt:1]: Do we need this still?

**Review by proegssilb — CHANGES_REQUESTED** [review]
If we don't need `requirements.txt` in the CI pipeline, I'd like to not have it in the PR.

File: requirements.txt
Comment (ultrabear) [reply requirements.txt:1]: This enables pip to cache poetry, it saves a few seconds in CI

File: .github/workflows/format.yml (hunk 1 above)
Comment (ultrabear) [reply .github/workflows/format.yml:3]: it will run on every push, so yes

**Review by proegssilb — APPROVED** [review]

**Pull request merged**